### ISSUE TITLE
Defining new ace rules

### ISSUE THREAD

**Issue by larsvoigt** [daisy/ace#84]
Is it on focus that ace user can define own epub specific rules like axe rules engine?

**Comment by rdeltour** [comment]
> Is it on focus that ace user can define own epub specific rules like axe rules engine?

Yes, this is definitely on our radar, although I'm not sure this will be implemented for 1.0.0 or a later version. Thanks for asking, this issue can now be used to track the progress :-)

**Comment by laudrain** [comment]
I'd like to add a custom rule like:
When <img> tag has role-"doc-cover", alt attribute must to start with "Couverture :" and should not end with ".jpg" or "png" string.
Do you think that kind of rule can be added to Ace with this mechanism?

**Comment by clapierre** [comment]
Luc, for one that would have to be localized so in English we would want "cover:..." 
But why is .jpg or .png not allowed for cover images, what should be used?

**Comment by clapierre** [comment]
I see this would be a end-user definable ruleset, that makes sense so then you can look for what you want.  I like that.

**Comment by rdeltour** [comment]
> Do you think that kind of rule can be added to Ace with this mechanism?

Yes, that would totally be the kind of logic that could be done with a custom rule. In fact this could be a good test bed for when we do this.

**Comment by laudrain** [comment]
@clapierre the « .jpg » or « .png » string should not be inside the alt attribute. The idea is to prevent alt text for the cover image containing « cover.jpg » !

@rdeltour will it be in vNext?

**Comment by clapierre** [comment]
@laudrain oh right, I thought you meant the image itself couldn't be of those types.  Yeah totally agree, that could be a check actually for any image's alt should never contain a .jpg / .png / .svg etc. and could even check for alt="image" and some of the other common things folks put in as a substitute for meaningful alt text.

**Comment by rdeltour** [comment]
> @rdeltour will it be in vNext?

No. This is a more advanced (and less critical) feature, that needs more time to mature.

**Comment by jenstroeger** [comment]
@laudrain, I think what you’d like to say is that the `alt` attribute should not reference a file, correct? Textual pattern check for `*.jpg` or some such would be too flakey, I think.

> Provide a hook to define custom Ace rules

…makes me wonder why not integrate them into Ace in the first place? To handle l10n better?

**Comment by laudrain** [comment]
@jenstroeger yes, anything the looks like a path to a file should rejected.

**Comment by rdeltour** [comment]
> why not integrate them into Ace in the first place?

there will always be the need for rules that do not necessarily implement a general a11y issue, but that implements a specific policy (organization/country/whatever).
